### PR TITLE
configuring travis-ci tests

### PR DESCRIPTION
**Pull request by daks**
Still WIP

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  - bundle update
+  # Setup for tests
+  - bundle exec kitchen create -l warn -c 4
+  # Provision containers
+  - bundle exec kitchen converge
+  # Run test-kitchen with docker driver:
+  - bundle exec kitchen verify
+after_script:
+  - bundle exec kitchen list
+  - bundle exec kitchen destroy
```
Comment (javierbertoli) [inline .travis.yml]: Hi @daks, perhaps you can use the files in the [packages formula](https://github.com/saltstack-formulas/packages-formula/) or the [apache formula](https://github.com/saltstack-formulas/apache-formula/) as a reference?

Ie, the [.travis.yml file](https://github.com/saltstack-formulas/apache-formula/blob/master/.travis.yml) can be greatly simplified and little is needed to get it up and running.

File: .kitchen.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+---
+driver:
+  name: docker
+
+driver_config:
+  use_sudo: false
+  privileged: true
+  provision_command: mkdir -p /run/sshd
+
```
Comment (javierbertoli) [inline .kitchen.yml:10]: You need to add

```yaml
  run_command: /lib/systemd/systemd
```

so the `service.sls` state can run.

File: .kitchen.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+provisioner:
+  name: salt_solo
+  log_level: info
+  require_chef: false
+  salt_install: bootstrap
+  salt_version: latest
+  salt_bootstrap_url: https://bootstrap.saltstack.com
+  salt_bootstrap_options: -X -p git -p curl -p sudo
+  is_file_root: true
```
Comment (javierbertoli) [inline .kitchen.yml]: You need to remove this line. It defines the root path for this formula, and it's at odds with the path defined in `state_top` below (that's why it's not finding the states).

File: .kitchen.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+  formula: logrotate
+  require_chef: false
+  require_chef: false
```
Comment (javierbertoli) [inline .kitchen.yml]: This `require_chef: false` line is 3 times, these 2 can be removed.

File: .kitchen.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        - logrotate.jobs
+
+verifier:
+  name: inspec
+#  sudo: false
+  # cli, documentation, html, progress, json, json-min, json-rspec, junit
+#  reporter: cli
+#  inspec_tests:
+#    - path: test/integration/default
+
```
Comment (javierbertoli) [inline .kitchen.yml:47]: You can re-enable the tests here, with

```yaml
verifier:
  name: inspec
  reporter:
    - cli
  inspec_tests:
    - path: test/integration/default
```

**Review by javierbertoli — CHANGES_REQUESTED** [review]
@daks, I tried the changes I'm suggesting locally and they worked. I think that with those we can get this working. Would you submit them when you have some time?

**Comment by daks** [comment]
@javierbertoli thanks to your help (and more available time), I pushed new code to add travis tests. Those work now, but tests on CentOS fail :(

**Comment by javierbertoli** [comment]
Checking locally, it seems RedHat family does not use a systemd timer to run logrotate, but a plain old cron, so the `service` state fails. 

It seems this PR is already finding bugs in the formula :smile: 

I think we'd probably need a conditional on that state file, to run only on Debian family and eventually add tests for `service running` / `cron existence` depending on the family?

wdyt?

**Comment by daks** [comment]
> It seems this PR is already finding bugs in the formula smile

and this is great!

> I think we'd probably need a conditional on that state file, to run only on Debian family and eventually add tests for service running / cron existence depending on the family?

I'm not fan of conditionals in the state, but maybe there is not better option. Eventually it can be activated with an option defined (depending of OS) in the map.jinja.

In any case, I'm not sure I'll be able to look at it soon so don't you think it would be better to merge this PR (eventually without the faulty test) and create an issue for later?

**Comment by javierbertoli** [comment]
> I'm not fan of conditionals in the state, but maybe there is not better option. Eventually it can be activated with an option defined (depending of OS) in the map.jinja.

Neither am I, but there are a few other formulas with states that are apply depending on the OS, as some OS differ this much (ie, systemd timer vs. cron), so I think in this case it would be the quickest solution.

> In any case, I'm not sure I'll be able to look at it soon so don't you think it would be better to merge this PR (eventually without the faulty test) and create an issue for later?

Problem is, what's failing is a state file, not a test. The formula is not applying completely because that state fails. I'll try to get a few minutes later today, modify these things we're discussing and probably merge this PR + push another PR to fix that part of the formula. Are you OK with this?

**Comment by daks** [comment]
I'm ok.

**Comment by javierbertoli** [comment]
Where I said 'a few minutes later today' should have read 'these days' but well... merged. :smile: 
Thanks, @daks.

**Pull request merged**